### PR TITLE
Feature: XOR folding

### PR DESCRIPTION
**Pull request by nbgl**
As discussed in #32. This implementation adds an extra argument to the CLI, and also an extra argument to the API, permitting the user to specify the number of folds.

### CODE REVIEW

**Comment by nbgl** [comment]
I've put the XOR folding in its own function, and changed the tests accordingly. I've also got a small test to check the integration of the XOR folding into existing code.

I would eventually like to include information such as the number of folds, as well as the number of hashes per field and the Bloom filter length, in the schema. It will make more sense to come up with a nice way to structure the hashing settings then. I've opened issue #45 for that.

File: clkhash/clk.py [hunk 1]
```diff
@@ -21,9 +21,10 @@
 log = logging.getLogger('clkhash.clk')
 
 
-def hash_and_serialize_chunk(chunk_pii_data,    # type: Iterable[Tuple[Any]]
-                             schema_types,      # type: Iterable[IdentifierType]
-                             keys               # type: Tuple[Tuple[bytes, ...], Tuple[bytes, ...]]
+def hash_and_serialize_chunk(chunk_pii_data, # type: Iterable[Tuple[Any]]
+                             schema_types,   # type: Iterable[IdentifierType]
+                             keys,           # type: Tuple[Tuple[bytes, ...], Tuple[bytes, ...]]
+                             xor_folds       # type: int
```
Comment (wilko77) [inline clkhash/clk.py]: corresponding docstring needs to be updated with :param xor_folds

**Comment by nbgl** [comment]
XOR folding.

**Comment by hardbyte** [comment]
Can you bump the version to `0.9.0` and update the `CHANGELOG.md` please @nbgl

**Comment by nbgl** [comment]
On it!

**Pull request merged**